upload/tests/views/test_uploads.py

```diff
@@ -288,6 +291,60 @@ def test_uploads_post(mock_metrics, db, mocker, mock_redis):
     upload_task_mock.assert_called()
 
 
+@override_settings(SHELTER_SHARED_SECRET="shelter-shared-secret")
+def test_uploads_post_shelter(db, mocker, mock_redis):
+    mocker.patch.object(
+        CanDoCoverageUploadsPermission, "has_permission", return_value=True
+    )
+    presigned_put_mock = mocker.patch(
+        "services.archive.StorageService.create_presigned_put",
+        return_value="presigned put",
+    )
+    upload_task_mock = mocker.patch(
+        "upload.views.uploads.UploadViews.trigger_upload_task", return_value=True
+    )
+
+    repository = RepositoryFactory(
+        name="the_repo", author__username="codecov", author__service="github"
+    )
+    commit = CommitFactory(repository=repository)
+    commit_report = CommitReport.objects.create(commit=commit, code="code")
+    repository.save()
+    commit_report.save()
+
+    owner = repository.author
+    client = APIClient()
+    client.force_authenticate(user=owner)
+    url = reverse(
+        "new_upload.uploads",
+        args=[
+            "github",
+            "codecov::::the_repo",
+            commit.commitid,
+            commit_report.code,
+        ],
+    )
+    response = client.post(
+        url,
+        {
+            "state": "uploaded",
+            "flags": ["flag1", "flag2"],
+            "version": "version",
+            "storage_path": "shelter/test/path.txt",
+        },
+        headers={
+            "X-Shelter-Token": "shelter-shared-secret",
+        },
+    )
+    upload = ReportSession.objects.filter(
+        report_id=commit_report.id, upload_extras={"format_version": "v1"}
+    ).first()
+    assert response.status_code == 201
```

Comment:
I'm confused: the other code I see is being gated by `not self.is_shelter_request()`, but this endpoint works. Is this intended or did I miss something? As far as I understand this, the existing logic should only work if it is not a shelter request

Reply:
The logic within `not self.is_shelter_request()` is the default behavior of generating the storage path in the API.  Only requests from Shelter are allowed to pass in their own storage path.